### ISSUE TITLE
How to modify the reported error

### ISSUE THREAD

**Issue by wymxz** [ofirpress/sandwich_transformer#2]
UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
/home/zqw/anaconda3/envs/pytorch1.5/lib/python3.7/site-packages/torch/optim/lr_scheduler.py:143: UserWarning: The epoch parameter in `scheduler.step()` was not necessary and is being deprecated where possible. Please use `scheduler.step()` to step the scheduler. During the deprecation, if epoch is different from None, the closed form is used instead of the new chainable form, where available. Please open an issue if you are unable to replicate your use case: https://github.com/pytorch/pytorch/issues/new/choose.
  warnings.warn(EPOCH_DEPRECATION_WARNING, UserWarning)

**Comment by wymxz** [comment]
Can it run on pytorch1.5 version?

**Comment by ofirpress** [comment]
Hi!
This is just a warning, it is not an error. I've discussed this with Sainbayar Sukhbaatar (the author of the original code base) and we both believe that this is a very minor issue that does not need to be fixed.

I've tested this code only on pytorch 1.2 (where I get the same warning). If, when using pytorch 1.5, you only get this warning, then I think everything is OK and you can keep on using pytorch 1.5. 

Tell me if you have more questions
